src/main/java/iog/psg/cardano/jpi/CardanoApi.java

```diff
@@ -62,63 +111,175 @@ public CardanoApi(iog.psg.cardano.CardanoApi api, HelpExecute helpExecute) {
+     * @throws NullPointerException
+     */
     public CompletionStage<CardanoApiCodec.Wallet> getWallet(
-            String fromWalletId) throws CardanoApiException {
+            String fromWalletId) throws CardanoApiException, NullPointerException {
 
         return helpExecute.execute(
                 api.getWallet(fromWalletId));
     }
 
+    /**
+     * Delete wallet by id
+     * Api Url: <a href="https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/deleteWallet">#deleteWallet</a>
+     *
+     * @param fromWalletId wallet's id
+     * @return void
+     * @throws CardanoApiException
+     * @throws NullPointerException
+     */
     public CompletionStage<Void> deleteWallet(
             String fromWalletId) throws CardanoApiException {
 
         return helpExecute.execute(
                 api.deleteWallet(fromWalletId)).thenApply(x -> null);
     }
 
+    /**
+     * Get transaction by id.
+     * Api Url: <a href="https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/getTransaction">#getTransaction</a>
+     *
+     * @param walletId wallet's id
+     * @param transactionId transaction's id
+     * @return get transaction request
+     */
     public CompletionStage<CardanoApiCodec.CreateTransactionResponse> getTransaction(
             String walletId, String transactionId) throws CardanoApiException {
 
         return helpExecute.execute(
                 api.getTransaction(walletId, transactionId));
     }
 
+    /**
+     * Estimate fee for the transaction. The estimate is made by assembling multiple transactions and analyzing the
+     * distribution of their fees. The estimated_max is the highest fee observed, and the estimated_min is the fee which
+     * is lower than at least 90% of the fees observed.
+     * Api Url: <a href="https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/postTransactionFee">#estimateFee</a>
+     *
+     * @param walletId wallet's id
+     * @param payments A list of target outputs ( address, amount )
+     * @return estimatedfee response
+     */
     public CompletionStage<CardanoApiCodec.EstimateFeeResponse> estimateFee(
             String walletId, List<CardanoApiCodec.Payment> payments) throws CardanoApiException {
         return estimateFee(walletId, payments, "self");
     }
 
+    /**
+     * Estimate fee for the transaction. The estimate is made by assembling multiple transactions and analyzing the
+     * distribution of their fees. The estimated_max is the highest fee observed, and the estimated_min is the fee which
+     * is lower than at least 90% of the fees observed.
+     * Api Url: <a href="https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/postTransactionFee">#estimateFee</a>
+     *
+     * @param walletId wallet's id
+     * @param payments A list of target outputs ( address, amount )
+     * @param withdrawal Optional, when provided, instruments the server to automatically withdraw rewards from the source
```

Comment:
All reference to 'withdrawal' need to say the value can be self or null, it makes no sense to say Optional in the jpi.